com.cubrid.cubridmigration.core/src/com/cubrid/cubridmigration/cubrid/CUBRIDSQLHelper.java

```diff
@@ -176,6 +178,35 @@ public String getDBQualifier(String objectName) {
         return getQuotedObjName(objectName);
     }
 
+    /**
+     * If the referential action is 'SET NULL', check the column and change it to 'RESTRICT' if it
+     * is 'NOT NULL'
+     *
+     * @param referentialAction int
+     * @param fk FK
+     * @return int
+     */
+    private int changeReferentialAction(int referentialAction, FK fk) {
+        if (referentialAction != 2) {
+            return referentialAction;
+        }
+
+        List<String> fkColumnNames = fk.getColumnNames();
+        Map<String, Column> tableColumnMap = new HashMap<String, Column>();
+        for (Column column : fk.getTable().getColumns()) {
+            tableColumnMap.put(column.getName(), column);
+        }
+
+        for (String columnName : fkColumnNames) {
+            if (!tableColumnMap.get(columnName).isNullable()) {
+                fk.setChangedReferentialAction(true);
+                return 1;
```

Comment:
I think it would be better to use defined variables.

Reply:
It returns the index value of the array(`UPDATE_RULE`, `DELETE_RULE`) and it is questionable whether a variable is needed to express this integer.
I'm curious about your opinion on this problem.

Reply:
> It returns the index value of the array(`UPDATE_RULE`, `DELETE_RULE`) and it is questionable whether a variable is needed to express this integer. I'm curious about your opinion on this problem.

UPDATE_RULE and DELETE_RULE are expressed as follows. 
==============================================
private static final String[] UPDATE_RULE =
            new String[] {"CASCADE", "RESTRICT", "SET NULL", "NO ACTION"};
    private static final String[] DELETE_RULE =
            new String[] {"CASCADE", "RESTRICT", "SET NULL", "NO ACTION"};
==============================================
Wouldn't it be possible to express the value int constants?
=========================
e.g)
private static final int CASCADE = 0
private static final int RESTRICT = 1
private static final int SET NULL = 2
private static final int NO_ACTION = 3
=========================

There is a function description, but it would be better if you could understand it through the code.

In fact, I could also change the constant in the FK class to one, but it may require a lot of changes, so you can define a new one in the class you use.

Reply:
I think it's clearer to use the constant as a variable rather than as it is.
I modified the code to make it a little simpler using a enum.